Add OCI artifact platform support (#1658)
kubernetes-sigs/security-profiles-operator: installation-usage.md

possible to add custom annotations to the security profile by using the
`--annotations` / `-a` flag multiple times in `KEY:VALUE` format.

### Using multiple platforms

`spoc push` supports specifying the target platforms for the profiles to be
pushed. This can be done by using the `--platforms` / `-p` together with the

> `--profiles` / `-p` flag. For example, to push two profiles into once artifact:

Reviewer: ```suggestion
`--profiles` / `-p` flag. For example, to push two profiles into one artifact:
```

  "config": {
    "mediaType": "application/vnd.unknown.config.v1+json",
    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
    "size": 2
  },
  "layers": [
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar",
      "digest": "sha256:6ddecdf312758a19ec788c3984418541274b3c9daf2b10f687d847bc283b391b",
      "size": 1167,
      "annotations": {
        "org.opencontainers.image.title": "profile-linux-arm64.yaml"
      },
      "platform": {
        "architecture": "arm64",
        "os": "linux"
      }
    },
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar",
      "digest": "sha256:6ddecdf312758a19ec788c3984418541274b3c9daf2b10f687d847bc283b391b",
      "size": 1167,
      "annotations": {
        "org.opencontainers.image.title": "profile-linux-amd64.yaml"
      },
      "platform": {
        "architecture": "amd64",
        "os": "linux"
      }
    }
  ],
  "annotations": {
    "org.opencontainers.image.created": "2023-04-28T08:59:17Z"
  }
}
```

There are a few fallback scenarios included in the CLI:

- If neither a platform nor a input file is specified, then `spoc` will fallback

>   to the default profile (`/tmp/profile.yaml`) and platform
>   (`runtime.GOOS`/`runtime.GOARCH`)

Reviewer: ```suggestion
  to the default profile (`/tmp/profile.yaml`) and platform
  (`runtime.GOOS`/`runtime.GOARCH`).
```

- If only one platform is specified, then `spoc` will apply it and use the
  default profile.
- If only one input file is specified, then `spoc` will apply it and use the
  default platform.
- If multiple platforms and input files are provided, then `spoc` requires them
  to match their occurrences. Platforms have to be unique as well.

The Security Profiles Operator will try to pull the correct profile by using
`runtime.GOOS`/`runtime.GOARCH`, but also falls back to the default profile

> (without any platform specified), if existing. `spoc pull` behaves in the same

Reviewer: ```suggestion
(without any platform specified), if it exists. `spoc pull` behaves in the same
```
